src/main/java/com/google/vinet/data/RegisteredUser.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.vinet.data;
+
+public class RegisteredUser {
+  private final String userId;
+
+  public RegisteredUser(String userId) {
+    this.userId = userId;
```

Comment:
A quick check here for a null `userId` being passed might help to prevent errors arising in other methods (for example,  `hashCode()` could be called later and will throw a `NullPointerException`, but this isn't too meaningful, since it wasn't `hashCode()` that introduced the necessary conditions for the exception to be thrown.)

Reply:
This is true. I have not strengthened the null checks on various data constructors.